src/Agent.Listener/Agent.cs

```diff
@@ -464,6 +465,17 @@ private async Task<int> RunAsync(AgentSettings settings, bool runOnce = false)
                                 Trace.Error($"Received message {message.MessageId} with unsupported message type {message.MessageType}.");
                             }
                         }
+                        catch (Exception e)
+                        {
+                            if (e is AggregateException)
```

Comment:
Can we move condition to the common function too?

Reply:
No, statement ELSE can have different set of commands like "Trace.Debug", "execution.Context.Error", "throw", etc. So I think it's better to keep IF statement outside of handler.